metric-publishers/emf-metric-publisher/src/test/java/software/amazon/awssdk/metrics/publishers/emf/internal/MetricEmfConverterTest.java

```diff
@@ -0,0 +1,178 @@
+                                                                                       Stream.of(HttpMetric.HTTP_CLIENT_NAME)
+                                                                                             .collect(Collectors.toSet()),
+                                                                                       Stream.of(MetricCategory.HTTP_CLIENT)
+                                                                                             .collect(Collectors.toSet()), MetricLevel.TRACE);
+    private final Clock fixedClock = Clock.fixed(
+        Instant.ofEpochMilli(12345678),
+        ZoneOffset.UTC
+    );
+
+    @BeforeEach
+    void setUp() {
+        metricEmfConverterDefault = new MetricEmfConverter(testConfigDefault,fixedClock);
+        metricEmfConverterCustom = new MetricEmfConverter(testConfigCustom, fixedClock);
+    }
+
+    @Test
+    void ConvertMetricCollectionToEMF_EmptyCollection() {
+        List<String> emfLogs = metricEmfConverterDefault.convertMetricCollectionToEmf(MetricCollector.create("test").collect());
+        for (String emfLog : emfLogs) {
+            assertThat(emfLog).isEqualTo("{\"_aws\":{\"Timestamp\":12345.678,\"LogGroupName\":\"my_log_group_name\","
+                                         + "\"CloudWatchMetrics\":[{\"Namespace\":\"AwsSdk/Test/JavaSdk2\",\"Dimensions\":[[]],"
+                                         + "\"Metrics\":[]}]}}");
+        }
+    }
+
+    @Test
+    void ConvertMetricCollectionToEMF_MultipleMetrics(){
+
+        MetricCollector metricCollector = MetricCollector.create("test");
+        metricCollector.reportMetric(HttpMetric.AVAILABLE_CONCURRENCY, 5);
+        metricCollector.reportMetric(HttpMetric.CONCURRENCY_ACQUIRE_DURATION, java.time.Duration.ofMillis(100));
+        List<String> emfLogs = metricEmfConverterDefault.convertMetricCollectionToEmf(metricCollector.collect());
+
+        for (String emfLog : emfLogs) {
+            assertThat(emfLog).isEqualTo("{\"_aws\":{\"Timestamp\":12345.678,\"LogGroupName\":\"my_log_group_name\",\"CloudWatchMetrics\":[{\"Namespace\":\"AwsSdk/Test/JavaSdk2\",\"Dimensions\":[[]],\"Metrics\":[{\"Name\":\"AvailableConcurrency\"},{\"Name\":\"ConcurrencyAcquireDuration\",\"Unit\":\"Milliseconds\"}]}]},\"AvailableConcurrency\":5,\"ConcurrencyAcquireDuration\":100.0}");
+
+        }
+    }
+
+
+    @Test
+    void ConvertMetricCollectionToEMF_Dimensions(){
+
+        MetricCollector metricCollector = MetricCollector.create("test");
+        metricCollector.reportMetric(HttpMetric.HTTP_CLIENT_NAME, "apache-http-client");
+        metricCollector.reportMetric(HttpMetric.AVAILABLE_CONCURRENCY, 5);
+        metricCollector.reportMetric(HttpMetric.HTTP_STATUS_CODE, 404);
+        List<String> emfLogs = metricEmfConverterDefault.convertMetricCollectionToEmf(metricCollector.collect());
+
+        for (String emfLog : emfLogs) {
```

Comment:
If we only expect one log statement, let's just assert that instead of for loop. Same for other test cases.

```
assertThat(emfLogs).containsOnly(...)
```